### PR TITLE
chore: Precommit setup 

### LINKED ISSUE #43: Pre-commit
setup lint-staged and husky

### PR DESCRIPTION
**Pull request by AhmedEldessouki**
Resolves #43

### CODE REVIEW

**Comment by AhmedEldessouki** [comment]
Resolves #43

File: lint-staged.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  '*.+(js|jsx|ts|tsx)': ['yarn lint'],
+  // '*.+(js|jsx|ts|tsx)': ['eslint', 'yarn test:precommit'],
+  '*.+(js|jsx|json|yml|yaml|css|ts|tsx)': ['prettier --write'],
```
Comment (Aprillion) [inline lint-staged.config.js:4]: prettier should be included in eslint config in `yarn lint`, no need to repeat on those files, only on `json|yml|yaml|css`, no?

**Review by Aprillion — APPROVED** [review]

File: lint-staged.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  '*.+(js|jsx|ts|tsx)': ['yarn lint'],
+  // '*.+(js|jsx|ts|tsx)': ['eslint', 'yarn test:precommit'],
```
Comment (merodiro) [inline lint-staged.config.js:3]: shouldn't we run `eslint` only on the changed files to make it faster?

can you remove the commented out line?

Comment (Aprillion) [reply lint-staged.config.js:3]: but `yarn lint` will already run eslint, won't uncommenting just add *more* steps?

given that it's merged, IMHO feel free to test locally and make a new PR with whatever config will be the fastest while doing all needed checks 😂

Comment (merodiro) [reply lint-staged.config.js:3]: it wasn't merged when I made the comment 😅
I will check it locally later today and make another PR if it is faster

Comment (AhmedEldessouki) [reply lint-staged.config.js:3]: sorry there wasn't electricity 😅😅
also @merodiro I have no idea how to make eslint run on the changed files only. the commented out line wouldn't do any harm (as far as my experience go)

**Comment by JacobMGEvans** [comment]
Yeah, I merged it and created another issue for the related comments.

Well done! 😁🚀

File: lint-staged.config.js (hunk 1 above)
Comment (AhmedEldessouki) [reply lint-staged.config.js:4]: for yml its helpful because it check its format. that is why i kept it separate

**Pull request merged**